components/org.wso2.carbon.identity.scim2.common/src/test/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManagerTest.java

```diff
@@ -0,0 +1,1716 @@
+        }
+        assertEquals(expect, result);
+    }
+
+    @DataProvider(name= "dpCreateRolePositive")
+    public Object[][] dpCreateRolePositive() {
+
+        return new Object[][]{
+                {SAMPLE_VALID_ROLE_ID, SAMPLE_VALID_ROLE_NAME, SAMPLE_TENANT_DOMAIN, SUCCESS},
+                {SAMPLE_VALID_ROLE_ID2, "", SAMPLE_TENANT_DOMAIN, SUCCESS},
+                {SAMPLE_VALID_ROLE_ID, null, SAMPLE_TENANT_DOMAIN, SUCCESS},
+                {"", null, SAMPLE_TENANT_DOMAIN, SUCCESS},
+                {null, null, SAMPLE_TENANT_DOMAIN, SUCCESS},
+                {SAMPLE_VALID_ROLE_ID2, null, "", SUCCESS},
+                {"", "", SAMPLE_TENANT_DOMAIN, SUCCESS},
+        };
+    }
+
+    @Test(dataProvider = "dpCreateRolePositive")
+    public void testCreateRolePositive(String roleId, String roleDisplayName, String tenantDomain, String expect)
+            throws IdentityRoleManagementException, BadRequestException, CharonException {
+
+        Role role = getDummyRole(roleId, roleDisplayName);
+
+        when(mockRoleManagementService.addRole(anyString(), anyListOf(String.class), anyListOf(String.class),
+                anyListOf(String.class), anyString())).
+                thenReturn(new RoleBasicInfo(roleId, roleDisplayName));
+
+        SCIMRoleManager scimRoleManager = new SCIMRoleManager(mockRoleManagementService, tenantDomain);
+
+        String result = "";
+        try {
+            Role createdRole = scimRoleManager.createRole(role);
+            assertEquals(createdRole.getDisplayName(), roleDisplayName);
+            assertEquals(createdRole.getId(), roleId);
+            result = SUCCESS;
+        }
+        catch (ConflictException e) {
+            if(e.getDetail().equals("Role with name: " + roleDisplayName +
+                    " already exists in the tenantDomain: "+ tenantDomain)) {
+                result = CONFLICT;
+            }else if(e.getDetail().equals("Role already exist for the role name: " + roleDisplayName)){
+                result = CONFLICT;
+            }
+        }
+        catch (BadRequestException e) {
+            result = BAD_REQUEST;
+        }
+        catch (CharonException e) {
+            result = UNEXPECTED_ERROR;
+        }
+        assertEquals(expect, result);
+    }
+
+    @DataProvider(name = "dpGetRoleNotFound")
+    public Object[][] dpGetRoleNotFound() {
+
+        return new Object[][]{
+                {SAMPLE_NON_EXISTING_ROLE_ID, SAMPLE_VALID_ROLE_NAME, "roleDomain1", SAMPLE_TENANT_DOMAIN, "attribute",
+                        true, FAIL},
+                {SAMPLE_NON_EXISTING_ROLE_ID2, SAMPLE_VALID_ROLE_NAME2, "roleDomainX", SAMPLE_INVALID_TENANT_DOMAIN,
+                        "attribute", false, FAIL}
+        };
+    }
+
+    @Test(dataProvider = "dpGetRoleNotFound")
+    public void testGetRoleNotFound(String roleId, String roleName, String domain, String tenantDomain,
+                                    String attribute, boolean attributeValue,
```

Comment:
attributeValue is not actually the value, right?

Reply:
to call the getRole function, we need to provide a Map<String, Boolean>. Hence I create a dummy Map object using those attribute values.